### PR TITLE
major/PLAT-866/add-otm-v2

### PR DESCRIPTION
**Pull request by zhelinisivanesan**

### CODE REVIEW

File: satellitevu/apis/otm.py [hunk 1]
```diff
@@ -233,3 +243,274 @@ def get_price(
...
+        **kwargs,
+    ):
+        """
+        Creates a tasking feasibility request.
+
+        Args:
+            contract_id: Associated ID of the Contract under which the feasibility
+            request will be performed.
+
+            coordinates: An array of coordinates.
```
Comment (DFEvans) [inline satellitevu/apis/otm.py:274]: Doesn't include the little addition I made yesterday :) https://github.com/SatelliteVu/satellitevu-client-python/pull/46

Applies to several other docstrings below

File: satellitevu/apis/otm.py [hunk 2]
```diff
@@ -233,3 +243,274 @@ def get_price(
...
+
+            date_from: datetime representing the start date of the feasibility request.
+
+            date_to: datetime representing the end date of the feasibility request.
+
+            day_night_mode: The mode of data capture, defaults to "day-night".
```
Comment (DFEvans) [inline satellitevu/apis/otm.py:280]: Would be good to document what possible values are allowed. `["day", "night", "day-night"]`?

Again, applies below to other docstrings.

File: satellitevu/apis/otm.py [hunk 3]
```diff
@@ -233,3 +243,274 @@ def get_price(
...
+    https://api.satellitevu.com/otm/v2/docs.
+    """
+
+    api_path = "otm/v2"
+    scopes = ["tasking:w"]
+
+    def post_feasibility(
+        self,
+        *,
+        contract_id: UUID,
```
Comment (c-wygoda) [inline satellitevu/apis/otm.py:260]: let's use `UUDI|str`, many people will just store their contract ID as a simple string.

File: satellitevu/apis/otm.py (hunk 1 above)
Comment (zhelinisivanesan) [reply satellitevu/apis/otm.py]: forgot that rebase 🤦‍♀️

File: satellitevu/apis/otm.py [hunk 4]
```diff
@@ -22,20 +24,22 @@ def post_feasibility(
...
+            coordinates: An array of coordinates - (longitude, latitude) or
+            (longitude, latitude, altitude).
 
             date_from: datetime representing the start date of the feasibility request.
 
             date_to: datetime representing the end date of the feasibility request.
 
-            day_night_mode: The mode of data capture, defaults to "day-night".
+            day_night_mode: String representing the mode of data capture. Allowed values
+            include ["day", "night", "day-night"]. Defaults to "day-night".
```
Comment (DFEvans) [inline satellitevu/apis/otm.py:42]: "Allowed values *include*" suggests to me there are more. It's just "Allowed values *are*", right?

File: satellitevu/apis/otm.py [hunk 5]
```diff
@@ -57,7 +61,12 @@ def post_feasibility(
         response = self.make_request(
             method="POST", url=url, json={k: v for k, v in payload.items() if v}
         )
-        return response.json()
+
+        if response.status != 202:
+            raise Exception(f"Error - {response.status} : {response.text}")
+
+        response_bytes = raw_response_to_bytes(response)
+        return loads(response_bytes.read())
```
Comment (DFEvans) [inline satellitevu/apis/otm.py:69]: What's the difference between jumping through these hoops and just `response.json()`?

File: satellitevu/client.py [hunk 6]
```diff
@@ -3,19 +3,21 @@
...
+from satellitevu.apis.otm import OtmV1, OtmV2
 from satellitevu.auth import AbstractCache, Auth
 from satellitevu.config import GATEWAY
 from satellitevu.http import AbstractClient, UrllibClient
 
 
 class FutureApis:
-    otm: OtmV1
+    otm_v1: OtmV1
+    otm_v2: OtmV2
```
Comment (DFEvans) [inline satellitevu/client.py:14]: When will these be promoted to non-future APIs, out of interest?

File: satellitevu/apis/otm.py (hunk 4 above)
Comment (zhelinisivanesan) [reply satellitevu/apis/otm.py]: makes sense to me!

File: satellitevu/apis/otm.py (hunk 5 above)
Comment (zhelinisivanesan) [reply satellitevu/apis/otm.py]: I was having a JSON decoder problem but it seems that was symptomatic of an error that I must have fixed elsewhere

File: satellitevu/client.py (hunk 6 above)
Comment (zhelinisivanesan) [reply satellitevu/client.py:14]: Unaware of an official timeline but I assume at SCO or post-SCO when customer contracts in Salesforce are in full swing

**Review by DFEvans — APPROVED** [review]

**Pull request merged**